src/app/pages/questions/services/questions.service.ts

```diff
@@ -72,29 +76,13 @@ export class QuestionsService {
     }
   }
 
-  showESMSleepQuestion() {
-    // Note: First ESM will show sleep question
-    return new Date().getHours() <= 9
-  }
-
-  showESMRatingQuestion() {
-    // Note: Last ESM will show rating question
-    // TODO: Fix hardcoded values
-    return new Date().getHours() >= 19
-  }
-
-  isESM(title) {
-    return title === 'ESM'
-  }
-
   processQuestions(title, questions: any[]) {
-    if (this.isESM(title)) {
-      const length = questions.length
-      const first = this.showESMSleepQuestion() ? 0 : 1
-      const last = this.showESMRatingQuestion() ? length - 1 : length - 2
-      return questions.slice(first, last)
-    }
-    return questions
+    return this.getHiddenQuestions().then(res => {
+      if (!res[title]) return questions
+      const questionsToHide = res[title].split(',').map(i => Number(i))
+      if (questionsToHide.length == questions.length) return questions
```

Comment:
I would expect if all questions are excluded then it should return nothing (but here it is returning the whole questionnaire?)

Reply:
Yeah, I didn't want to support returning nothing because if the questionnaire shouldn't show up, it should be modified in the protocol instead? Because this is after pressing the start button.

Reply:
Ah i see. Ok thanks